src/dmd/backend/cgsched.d

```diff
@@ -2122,147 +2227,136 @@ if (c2->IEVpointer1 + sz2 <= c1->IEVpointer1) printf("t5\n");
     }
 
     // Get the instructions out of the schedule table
-    assert((unsigned)tblmax <= TBLMAX);
+    assert(cast(uint)tblmax <= TBLMAX);
     for (int i = 0; i < tblmax; i++)
     {
         Cinfo* ci = tbl[i];
-#ifdef DEBUG
+
+        debug
         if (debugs)
         {
             if (PRO)
-            {   static char tbl[3][4] = { "0  "," 1 ","  2" };
+            {   immutable char[4][3] tbl = [ "0  "," 1 ","  2" ];
 
                 if (ci)
-                    printf("%s %d ",tbl[i - ((i / 3) * 3)],ci->uops);
+                    printf("%s %d ",tbl[i - ((i / 3) * 3)].ptr,ci.uops);
                 else
-                    printf("%s   ",tbl[i - ((i / 3) * 3)]);
+                    printf("%s   ",tbl[i - ((i / 3) * 3)].ptr);
             }
             else
             {
                 printf((i & 1) ? " V " : "U  ");
             }
             if (ci)
-                ci->c->print();
+                ci.c.print();
             else
                 printf("\n");
         }
-#endif
+
         if (!ci)
             continue;
-        fpustackused += ci->fpuadjust;
+        fpustackused += ci.fpuadjust;
         //printf("stage()1: fpustackused = %d\n", fpustackused);
-        c = ci->c;
+        c = ci.c;
         if (i == 0)
-            c->Iflags |= CFtarg;        // by definition, first is always a jump target
+            c.Iflags |= CFtarg;        // by definition, first is always a jump target
         else
-            c->Iflags &= ~CFtarg;       // the rest are not
+            c.Iflags &= ~CFtarg;       // the rest are not
 
         // Put in any FXCH prefix
-        if (ci->fxch_pre)
+        if (ci.fxch_pre)
         {   code *cf;
             assert(i);
-            cf = gen2(NULL,0xD9,0xC8 + ci->fxch_pre);
+            cf = gen2(null,0xD9,0xC8 + ci.fxch_pre);
             *pc = cf;
-            pc = &code_next(cf);
+            pc = &cf.next;
         }
 
         *pc = c;
         do
         {
             assert(*pc != code_next(*pc));
-            pc = &code_next(*pc);
+            pc = &(*pc).next;
         } while (*pc);
 
         // Put in any FXCH postfix
-        if (ci->fxch_post)
+        if (ci.fxch_post)
         {
             for (int j = i + 1; j < tblmax; j++)
             {   if (tbl[j])
-                {   if (tbl[j]->fxch_pre == ci->fxch_post)
+                {   if (tbl[j].fxch_pre == ci.fxch_post)
                     {
-                        tbl[j]->fxch_pre = 0;           // they cancel each other out
+                        tbl[j].fxch_pre = 0;           // they cancel each other out
                         goto L1;
                     }
                     break;
                 }
             }
             {   code *cf;
-                cf = gen2(NULL,0xD9,0xC8 + ci->fxch_post);
+                cf = gen2(null,0xD9,0xC8 + ci.fxch_post);
                 *pc = cf;
-                pc = &code_next(cf);
+                pc = &cf.next;
             }
         }
     L1: ;
```

Comment:
The newly added [DScanner](https://github.com/dlang-community/D-Scanner) (static source code analyzer and linter) complains about this empty declaration here.